hikari/api/rest.py

```diff
@@ -5047,6 +5047,107 @@ async def edit_welcome_screen(
+        """Fetch a guild's membership verification gate screen.
+
+        Parameters
+        ----------
+        guild : hikari.snowflakes.SnowflakeishOr[hikari.guilds.PartialGuild]
+            Object or ID of the guild to fetch the member verification screen for
+
+        Returns
+        -------
+        hikari.guilds.MembershipGate
+            The requested membership verification gate screen.
+
+        Raises
+        ------
+        hikari.errors.ForbiddenError
+            If you are not part of the guild or if you are missing the
+            "MANAGE_GUILD" permission or if the guild does not have access to the
+            feature (is not a community guild).
+        hikari.errors.NotFoundError
+            If the guild is not found.
+        hikari.errors.UnauthorizedError
+            If you are unauthorized to make the request (invalid/missing token).
+        hikari.errors.RateLimitTooLongError
+            Raised in the event that a rate limit occurs that is
+            longer than `max_rate_limit` when making a request.
+        hikari.errors.RateLimitedError
+            Usually, Hikari will handle and retry on hitting
+            rate-limits automatically. This includes most bucket-specific
+            rate-limits and global rate-limits. In some rare edge cases,
+            however, Discord implements other undocumented rules for
+            rate-limiting, such as limits per attribute. These cannot be
+            detected or handled normally by Hikari due to their undocumented
+            nature, and will trigger this exception if they occur.
+        hikari.errors.InternalServerError
+            If an internal error occurs on Discord while handling the request.
+        """
+
+    @abc.abstractmethod
+    async def edit_guild_membership_gate(
+        self,
+        guild: snowflakes.SnowflakeishOr[guilds.PartialGuild],
+        *,
+        description: undefined.UndefinedNoneOr[str] = undefined.UNDEFINED,
+        enabled: undefined.UndefinedOr[bool] = undefined.UNDEFINED,
+        fields: undefined.UndefinedOr[typing.Sequence[guilds.MembershipGateField]] = undefined.UNDEFINED,
+    ) -> guilds.MembershipGate:
+        """Edit the verification gate screen of a community guild.
+
+        Parameters
+        ----------
+        guild : hikari.snowflakes.SnowflakeishOr[hikari.guilds.PartialGuild]
+            ID or object of the guild to edit the welcome screen for.
+
+        Other Parameters
+        ----------------
+        description : hikari.undefined.UndefinedNoneOr[builtins.str]
+            The new description to set. This may be passed as `builtins.None` to
+            remove the set description or left as `hikari.undefined.UNDEFINED`
+            to leave the description unchanged.
+        enabled : hikari.undefined.UndefinedOr[builtins.bool]
+            Whether the member screening feature should be enabled or not.
+            Leave as `hikari.undefined.UNDEFINED` to leave unchanged.
```

Comment:
Should this "leave as default to leave it unchanged" be moved to the description? (also, 2 different styles here... gah... "leave as ..." and "left as ...")